File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/loseopportunity.vb
' <snippetloseopportunity>


 Dim loseOppRequest = New LoseOpportunityRequest With
                      {
                          .OpportunityClose =
                          New OpportunityClose With
                          {
                              .OpportunityId = New EntityReference(Opportunity.EntityLogicalName,
                                                                   _loseOpportunityId)},
                          .Status = New OptionSetValue(CInt(Fix(opportunity_statuscode.Canceled)))
                      }

 _serviceProxy.Execute(loseOppRequest)

 Console.WriteLine("Opportunity closed as Lost.")

' </snippetloseopportunity>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/loseopportunity.cs
// <snippetloseopportunity>


 var loseOppRequest = new LoseOpportunityRequest
 {
     OpportunityClose = new OpportunityClose
     {
         OpportunityId = new EntityReference
             (Opportunity.EntityLogicalName, _loseOpportunityId)
     },
     Status = new OptionSetValue((int)opportunity_statuscode.Canceled)
 };

 _serviceProxy.Execute(loseOppRequest);

 Console.WriteLine("Opportunity closed as Lost.");

// </snippetloseopportunity>